pandas/tests/frame/methods/test_fillna.py

```diff
@@ -57,7 +57,10 @@ def test_fillna_on_column_view(self, using_copy_on_write):
 
         # i.e. we didn't create a new 49-column block
         assert len(df._mgr.arrays) == 1
-        assert np.shares_memory(df.values, arr)
+        if using_copy_on_write:
+            assert not np.shares_memory(df.values, arr)
```

Comment:
Or pass copy=False to the constructor instead? 

Because now the check above about ``assert np.isnan(arr[:, 0]).all()`` is kind of useless because `arr` was copied and so of course will not be modified.

Reply:
No, we are doing df[0] which is a chained assignment case. Updated the test to set copy=False

Reply:
Ah, yes, missed the `[0]` ..